pkg/server/events_test.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"testing"
+	"time"
+
+	eventtypes "github.com/containerd/containerd/api/events"
+	"github.com/stretchr/testify/assert"
+)
+
+// TestBackOff tests the logic of backOff struct.
+func TestBackOff(t *testing.T) {
+	inputQueues := map[string]backOffQueue{
+		"container1": {
+			events: []interface{}{
+				&eventtypes.TaskExit{ContainerID: "container1", ID: "1"},
+				&eventtypes.TaskExit{ContainerID: "container1", ID: "2"},
+			},
+		},
+		"container2": {
+			events: []interface{}{
+				&eventtypes.TaskExit{ContainerID: "container2", ID: "1"},
+				&eventtypes.TaskExit{ContainerID: "container2", ID: "2"},
+			},
+		},
+	}
+	expectedQueues := map[string]backOffQueue{
+		"container2": {
+			events: []interface{}{
+				&eventtypes.TaskExit{ContainerID: "container2", ID: "1"},
+				&eventtypes.TaskExit{ContainerID: "container2", ID: "2"},
+			},
+		},
+		"container1": {
+			events: []interface{}{
+				&eventtypes.TaskExit{ContainerID: "container1", ID: "1"},
+				&eventtypes.TaskExit{ContainerID: "container1", ID: "2"},
+			},
+		},
+	}
+
+	t.Logf("Should can backOff a event")
+	actual := newEventMonitor(nil, nil).backOff
```

Comment:
? I prefer using the `new` function suggested in https://github.com/containerd/cri/pull/628/files#r174375426.